server/src/main/java/org/elasticsearch/plugins/PluginInfo.java

```diff
@@ -172,36 +183,45 @@ public static PluginInfo readFromProperties(final Path path) throws IOException
             extendedPlugins = Arrays.asList(Strings.delimitedListToStringArray(extendedString, ","));
         }
 
-        final String hasNativeControllerValue = propsMap.remove("has.native.controller");
-        final boolean hasNativeController;
-        if (hasNativeControllerValue == null) {
-            hasNativeController = false;
-        } else {
-            switch (hasNativeControllerValue) {
-                case "true":
-                    hasNativeController = true;
-                    break;
-                case "false":
-                    hasNativeController = false;
-                    break;
-                default:
-                    final String message = String.format(
-                            Locale.ROOT,
-                            "property [%s] must be [%s], [%s], or unspecified but was [%s]",
-                            "has_native_controller",
-                            "true",
-                            "false",
-                            hasNativeControllerValue);
-                    throw new IllegalArgumentException(message);
-            }
+        final boolean hasNativeController = parseBooleanValue("has.native.controller", propsMap.remove("has.native.controller"));
+
+        final String typeString = propsMap.remove("type");
+        if (Strings.isNullOrEmpty(typeString)) {
+            throw new IllegalArgumentException("property [type] is missing for plugin [" + name + "]");
```

Comment:
I think this will break a lot of plugin authors who have static plugin properties. Can we instead default type to isolated?

Reply:
It now defaults to isolated.